lapis2/src/main/kotlin/org/genspectrum/lapis/model/SiloQueryModel.kt

```diff
@@ -50,13 +50,23 @@ class SiloQueryModel(
             ),
         )
         return data.map {
-            val sequenceName =
-                if (referenceGenomeSchema.isSingleSegmented()) it.mutation else "${it.sequenceName}:${it.mutation}"
+            val mutation = if (referenceGenomeSchema.isSingleSegmented()) {
+                it.mutation
+            } else {
+                "${it.sequenceName}:${it.mutation}"
+            }
 
             NucleotideMutationResponse(
-                sequenceName,
-                it.count,
-                it.proportion,
+                mutation = mutation,
+                count = it.count,
+                proportion = it.proportion,
+                sequenceName = when (referenceGenomeSchema.isSingleSegmented()) {
+                    true -> null
```

Comment:
Is null correct here? Do we want to set it to that specific value?

Reply:
We want to specifically set `null`.